packages/metadata/src/inspector.ts

```diff
@@ -0,0 +1,218 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/metadata
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Reflector, NamespacedReflect} from './reflect';
+import {MetadataMap} from './decorator-factory';
+
+/**
+ * TypeScript reflector without a namespace. The TypeScript compiler can be
+ * configured to add design time metadata.
+ *
+ * See https://www.typescriptlang.org/docs/handbook/decorators.html
+ */
+const TSReflector = new NamespacedReflect();
+
+/**
+ * Design time metadata for a method
+ */
+export interface DesignTimeMethodMetadata {
+  type: Function;
+  parameterTypes: Function[];
+  returnType: Function;
+}
+
+/**
+ * Inspector for metadata applied by decorators
+ */
+export class MetadataInspector {
+  static readonly Reflector = Reflector;
+  static readonly DesignTimeReflector = TSReflector;
+
+  /**
+   * Get the metadata associated with the given key for a given class
+   * @param key Metadata key
+   * @param target Class that contains the metadata
+   * @param ownOnly Optional flag to control if only own metadata is inspected.
+   * The default value is `false` and inherited metadata is inspected.
+   */
+  static getClassMetadata<T>(
+    key: string,
+    target: Function,
+    ownOnly?: boolean,
```

Comment:
Sorry for not catching this earlier. Boolean arguments are an anti-pattern - how are the readers of the code invoking this method going to understand what `true` or `false` means?

```ts
const meta = getClassMetadata('mykey', myTarget, false);
```

Such API is also difficult to extend later, when we need to add more flags than just `ownOnly`.

I am proposing to change all methods accepting a boolean `ownOnly` argument to accept an "options" or "flags" object.

```ts
interface InspectOptions {
  ownOnly?: boolean
}

static getClassMetadata<T>(
  key: string,
  target: Function,
  options: InspectOptions = {})
```